src/database.rs

```diff
@@ -89,6 +88,21 @@ pub trait Database: Send + AsDynDatabase + Any + ZalsaDatabase {
     {
         crate::attach::attach(self, || op(self))
     }
+
+    #[doc(hidden)]
+    fn zalsa_register_downcaster(&self) {
+        // The no-op downcaster is special cased in view caster construction.
+    }
```

Comment:
The perf changes come from this being a no-op now, so they are kind of fake as you only get that benefit if you don't use a custom database trait

Reply:
should we update the benchmarks to use a custom
database? I assume any production use of Salsa would need a custom database to do the lazy input pattern to load, e.g., files.

Reply:
Lukas said: 

> unrelated to [this] PR I'd say, but I don't think that's relevant for the benchmarks either. The only difference is a no-op vs upcast (a dyn trait pointer reconstruction)